### ISSUE TITLE
Add equivalent of object-property-newline for import declaration

### ISSUE THREAD

**Issue by gajus** [benmosher/eslint-plugin-import#344]
[`object-property-newline`](http://eslint.org/docs/rules/object-property-newline)

> While formatting preferences are very personal, a number of style guides require that object properties be placed on separate lines for better readability.
> Another argument in favor of this style is that it improves the readability of diffs when a property is changed:

Examples of *incorrect* code for this rule:

```js
import {createMovieMatcher,matcher} from './../src/movieRecognizer';
```

Examples of *correct* code for this rule:

```js
import {
    createMovieMatcher,
    matcher
} from './../src/movieRecognizer';
```

**Comment by jfmengels** [comment]
Sounds good to me. Would prefer having an option like `"always"` / `"never"` on this rule, so that one could decide to go the opposite way if preferred.

**Comment by lo-corp** [comment]
Is this still a work in progress? This was posted in 2016

**Comment by ljharb** [comment]
yep.

**Comment by julekgwa** [comment]
try https://github.com/gmsorrow/eslint-plugin-modules-newline

**Comment by ljharb** [comment]
Given that exists as a model, it seems like someone could make a PR to add it here.

**Comment by TheAkio** [comment]
It would also be nice if you could have an `as-needed` option which takes the configured maximum line length (or you just configure it again with the rule) and puts it on one line if it fits within the line length but breaks it up into multiple lines if it doesn't.

There is a VSCode extension that made this feature pretty configurable so some things might be nice to implement into such an ESLint rule: https://marketplace.visualstudio.com/items?itemName=dozerg.tsimportsorter

**Comment by gajus** [comment]
The other plugin is archived now. I've implemented a rule that always forces a new line in Canonical plugin:

* [`export-specifier-newline`](https://github.com/gajus/eslint-plugin-canonical#eslint-plugin-canonical-rules-export-specifier-newline)
* [`import-specifier-newline`](https://github.com/gajus/eslint-plugin-canonical#eslint-plugin-canonical-rules-import-specifier-newline)

**Comment by ljharb** [comment]
Still happy to accept a PR here for it.
